python_hooks/validate_branch_name.py

```diff
@@ -0,0 +1,22 @@
+import re
+import sys
+from subprocess import check_output
+
+# regex patterns for acceptable branch names, should be in the format
+# JIRA-1234/string-with-dashes_or_underscores_or_numbers
+# hotfix/string-with-dashes_or_underscores_or_numbers (when no jira ticket exists)
+acceptable_branch_names = [r"^([a-zA-Z]+-\d+\/[a-zA-Z0-9_-]+$).*", r"^(hotfix\/[a-zA-Z0-9_-]+$).*"]
```

Comment:
I like this! Just hopefully Engineers don't flip out on the more tightly controlled branch name scheme. Alternatively, maybe just exclude the few special chars that are particularly problematic.

Reply:
should we also enforce a character limit?

Reply:
yeah, more proximal step of just limiting special characters is probably a better idea the more I think of it.

do we run into issues because of character limits?

Reply:
Oh... right:
 https://docs.docker.com/reference/cli/docker/image/tag/#:~:text=The%20tag%20must%20be%20valid,no%20longer%20than%20128%20characters.

Reply:
yeah, it can be an issue. we [try to solve downstream](https://github.com/tatari-tv/github-actions/blob/19a00d04f701ccd3df693bd8c2bb4447f636be1b/ephemeral-env/make/entrypoint.sh#L221) but we should also be training devs to stop using branch names, like commit messages.

As linked, prevent names longer than 51 chars for ephemeral FE apps, for example.

Reply:
yikes, in philo repo...
```
$ git branch -r|grep -c -E '.{52,}'
67
```

Reply:
yeesh, yeah... that's getting carried away